### PR TITLE
update Open MPI to 4.1.1

### PR DESCRIPTION
**Pull request by simonbyrne**
Close #2062

### CODE REVIEW

**Comment by giordano** [comment]
FreeBSD failure is the same as https://github.com/JuliaPackaging/Yggdrasil/pull/2062#issuecomment-727617390

**Comment by giordano** [comment]
Are you ok to require Julia v1.6 and build for the new platforms?

**Comment by simonbyrne** [comment]
Yeah, that's fine

File: O/OpenMPI/build_tarballs.jl [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
 # Install the library
-make "${flags[@]}" install
+make install
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line.
 #platforms = supported_platforms()
-platforms = filter(!Sys.iswindows, supported_platforms())
+platforms = filter(p -> !Sys.iswindows(p) && !(arch(p) == "armv6l" && libc(p) == "glibc"), supported_platforms(; experimental=true))
```
Comment (giordano) [inline O/OpenMPI/build_tarballs.jl:37]: It's quite interesting it's failing only for glibc with
```
[16:14:10] libtool: compile:  cc -std=gnu99 -DHAVE_CONFIG_H -I. -I../../../src/include -I../../../include -iquote/workspace/srcdir/openmpi-4.1.1/opal/mca/pmix/pmix3x/pmix -iquote/workspace/srcdir/openmpi-4.1.1/opal/mca/pmix/pmix3x/pmix/src -I/workspace/srcdir/openmpi-4.1.1/opal/mca/pmix/pmix3x/pmix/include -I/workspace/srcdir/openmpi-4.1.1 -I/workspace/srcdir/openmpi-4.1.1/opal/include -I/workspace/srcdir/openmpi-4.1.1/opal/mca/event/libevent2022/libevent -I/workspace/srcdir/openmpi-4.1.1/opal/mca/event/libevent2022/libevent/include -I/workspace/srcdir/openmpi-4.1.1/opal/mca/hwloc/hwloc201/hwloc/include -DNDEBUG -O3 -finline-functions -fno-strict-aliasing -pthread -MT base/plog_base_select.lo -MD -MP -MF base/.deps/plog_base_select.Tpo -c base/plog_base_select.c  -fPIC -DPIC -o base/.libs/plog_base_select.o
[16:14:10] /tmp/ccAfPlen.s: Assembler messages:
[16:14:10] /tmp/ccAfPlen.s:122: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:148: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:187: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:234: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:473: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:549: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:658: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:667: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:693: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:710: Error: selected processor does not support ARM mode `dmb'
[16:14:10] /tmp/ccAfPlen.s:748: Error: selected processor does not support ARM mode `dmb'
```
I'll try to look into this later, but for the time being let's skip it and move on.

**Pull request merged**